src/napi/node_api_value.c

```diff
@@ -483,3 +484,53 @@ napi_status napi_strict_equals(napi_env env, napi_value lhs, napi_value rhs,
 
   return napi_assign_bool(jerry_get_boolean_value(is_equal), result);
 }
+
+napi_status napi_create_promise(napi_env env, napi_deferred* deferred,
+                                napi_value* promise) {
+  jerry_value_t jpromise = jerry_create_promise();
```

Comment:
What happens when promise is not supported by the build? IMHO we should use `jerry_is_feature_enabled` in these functions.

Reply:
```c
  if (!jerry_is_feature_enabled(JERRY_FEATURE_PROMISE)) {
    NAPI_ASSIGN(result, NULL);
    NAPI_RETURN(napi_generic_failure,
                "Promise is not supported by this build.");
  }
```

Reply:
If promise is not supported JerryScript simply returns an error value.

Reply:
It looks like we have multiple choices of throwing an exception. I used the `napi_generic_failure` in #1855 (see my previous comment). It looks less costly then your method. Which one do you prefer?

Reply:
If you used that one I think we can stick to that, so it will be consistent. I'll update my PR